src/sets.jl

```diff
@@ -4,25 +4,51 @@
 # Use of this source code is governed by an MIT-style license that can be found
 # in the LICENSE.md file or at https://opensource.org/licenses/MIT.
 
-# Sets
-
 # Note: When adding a new set, also add it to Utilities.Model.
 
 """
     AbstractSet
 
-Abstract supertype for set objects used to encode constraints. A set object
-should not contain any [`VariableIndex`](@ref) or [`ConstraintIndex`](@ref)
-as the set is passed unmodifed during [`copy_to`](@ref).
+Abstract supertype for set objects used to encode constraints.
+
+!!! note
+    When developing new sets, the set struct must not contain any
+    [`VariableIndex`](@ref) or [`ConstraintIndex`](@ref) objects, because the
+    set is passed unmodifed during [`copy_to`](@ref).
```

Comment:
I don't think the reader will understand my a set needs to be modified during `copy_to` unless he's a core dev

Reply:
Removed the `because ... copy_to` clause, and moved the note to a `## Note for developers` since it doesn't apply to users.